crates/signer/src/service.rs

```diff
@@ -69,16 +73,40 @@ impl SigningService {
     }
 }
 
+/// Authentication middleware layer
+async fn jwt_auth(
+    State(state): State<SigningState>,
+    TypedHeader(auth): TypedHeader<Authorization<Bearer>>,
+    mut req: Request,
+    next: Next,
+) -> Result<Response, SignerModuleError> {
+    let jwt: Jwt = auth.token().to_string().into();
+
+    let maybe_module_id = state.jwts.get_by_right(&jwt);
+
+    let Some(module_id) = maybe_module_id else {
```

Comment:
nit:
```rust
let module_id = maybe_module_id.ok_or_else(|| {
    warn!("Unauthorized request. Was the module started correctly?");
    SignerModuleError::Unauthorized
})?;
```
Also if you return an error, the message should also be an error

Reply:
Anything particular we gain this way or is this a matter of taste preference?
I tend to prefer the glowy red (well, in my color scheme :smiley:) `return` keyword when we're on about early returning an error we're generating in the same place.

About the error message, I presume you mean `err!` instead of `warn!` as the trace level. The thing for me here is, the error we're returning is the opposite of 'ok' in rust's `Result` context, where there's no levels of errors. It's up to debate from then on whether an unauthorized request is an error or a warning from the server's perspective ('error' here in tracing level context). At the end of the day, I'm fine with whatever. :smiley:

Reply:
Fixed in f0b9078e590a12ff440470480df256b1f0e52b53